frontend/constants.ts

```diff
@@ -1 +1,33 @@
-export const baseUrl = process.env.GATSBY_BASE_URL || ''
+const baseUrl = process.env.GATSBY_BASE_URL
+
+export function getBaseUrl(): string {
+  if (baseUrl) {
+    return baseUrl
+  }
+
+  /*
+  This is a special case for production.
+
+  We want to be able to build the front end with no value set for
+  `GATSBY_BASE_URL` so that we can deploy a build to staging
+  and then promote the exact same build to production.
+
+  When deployed to staging, we want the frontend on
+  https://staging.shields.io/ to generate badges with the base
+  https://staging.shields.io/
+
+  When we promote to production we want https://shields.io/ and
+  https://www.shields.io/ to both generate badges with the base
+  https://img.shields.io/
+  */
+  try {
+    const { protocol, hostname } = window.location
+    if (['shields.io', 'www.shields.io'].includes(hostname)) {
+      return 'https://img.shields.io'
+    }
+    return `${protocol}//${hostname}`
+  } catch (e) {
+    // server-side rendering
+    return ''
```

Comment:
This is not ideal, but I'm not sure there is a better option available given the constraints

Reply:
I think instead of a `try` you can do `if (typeof window === 'undefined')`.

https://www.gatsbyjs.com/docs/debugging-html-builds/#how-to-check-if-window-is-defined

Reply:
:laughing: The thing I was commenting on here was `return ''` as opposed to the use of try/catch

It is neither here nor there whether we use a try/catch or an if/then: It is unfortunate that the server-rendered version of the page will not use `img.shields.io` as the base for badge images, but I don't see another option.

Reply:
Ahh gotcha, I misunderstood.

Yea, that we can't get the right URL for the server rendering is unfortunate. It's fortunate that this path is only used for badge previews.

I think there is an alternative here in the long run, which is that we wire something up, probably in Gatsby, to render the previews at build time and inline them.